### PR TITLE
SI-860 Complete UTF8 character set fix

### PR DESCRIPTION
**Pull request by uuf6429**

### CODE REVIEW

File: src/api/nl2go/index.php [hunk 1]
```diff
@@ -603,50 +616,18 @@ private function transformCustomerToRecipient($email, $status){
...
-    private function encodeToUtf8($value)
-    {
-        return mb_convert_encoding($value, 'utf-8', $GLOBALS['charset']);
-    }
 }
 
 $username = isset($_SERVER['PHP_AUTH_USER']) ? $_SERVER['PHP_AUTH_USER'] : '';
 $apikey = isset($_SERVER['PHP_AUTH_PW']) ? $_SERVER['PHP_AUTH_PW'] : '';
-$action = isset($_POST['action']) ? $_POST['action'] : '';
+$action = isset($_REQUEST['action']) ? $_REQUEST['action'] : '';
```
Comment (uuf6429) [inline src/api/nl2go/index.php:623]: **Reason:** The change here and below allows us to test simply by visiting the URL, eg:
> https://some-shop/api/nl2go/index.php?action=getCustomers&username=newsletter2go&apikey=6AAA57c53531d2b9fffd1479aac09ee8

File: src/api/nl2go/index.php [hunk 2]
```diff
@@ -603,50 +616,18 @@ private function transformCustomerToRecipient($email, $status){
...
         return $result;
     }
-
-    /**
-     * Checks global charset
-     *
-     * @param mixed $data
-     * @return string
-     */
-    private function checkEncoding($data)
```
Comment (uuf6429) [inline src/api/nl2go/index.php:613]: **Reason:** This function and the one below would have caused double-encoding.

File: src/api/nl2go/index.php [hunk 3]
```diff
@@ -151,12 +158,7 @@ public function getCustomers()
         $n = xtc_db_num_rows($customersQuery);
 
         for ($i = 0; $i < $n; $i++) {
-            $cs = xtc_db_fetch_array($customersQuery);
-            foreach ($cs as $key => $value) {
-                $cs[$key] = utf8_encode($value);
```
Comment (uuf6429) [inline src/api/nl2go/index.php:156]: **Reason:** This was causing the implementation to still work when using `latin1` for `getCustomGroups` (with good encoding) but failed for `getCustomers` (with bad encoding).

File: src/api/nl2go/index.php [hunk 4]
```diff
@@ -39,7 +39,12 @@ class N2GoApi
 
     public function __construct($action, $username, $apikey, $getParams = array(), $postParams = array())
     {
+        $response = [];
+
         try {
+            xtc_db_query('SET NAMES utf8');
+            xtc_db_query('SET CHARACTER SET utf8');
```
Comment (uuf6429) [inline src/api/nl2go/index.php]: **Reason:** And the unified fix goes here. 😀

**Comment by mimonewsletter2go** [comment]
can you update the version of the plugin too?

**Pull request merged**